File: my-code/242/main.go
package main

// @Time     : 2021/5/30 20:53
// @Author   : Ranshi
// @File     : 242. 有效的字母异位词.go
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}
	diff := make(map[uint8]int)
	for i := 0; i < len(s); i++ {
		if _, ok := diff[s[i]]; ok {
			diff[s[i]]++
		} else {
			diff[s[i]] = 1
		}
		if _, ok := diff[t[i]]; ok {
			diff[t[i]]--
		} else {
			diff[t[i]] = -1
		}
	}
	for _, x := range diff {
		if x != 0 {
			return false
		}
	}
	return true
}

File: my-code/242/main.py
# -*- coding: UTF-8 -*-
# @Time     : 2021/08/31 08:44
# @Author   : Ranshi
# @File     : 123.py


class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        for i in range(26):
            ch = chr(ord("a") + i)
            if s.count(ch) != t.count(ch):
                return False
        return True


if __name__ == "__main__":
    print(Solution().isAnagram(s="anagram", t="nagaram"))
